test/integration/users.test.js

```diff
@@ -0,0 +1,156 @@
+const supertest = require('supertest');
+const { factory } = require('factory-girl');
+
+const dbHelper = require('../helpers/db');
+const usersTestData = require('../data/users');
+const app = require('../../app');
+const { errorMessages } = require('../../app/constants');
+
+require('../setup');
+
+const postUsersPath = '/users';
+const defaultHeaders = ['Accept', 'application/json'];
+
+describe('Create users POST  /users', () => {
+  describe('Should work correctly with satisfactory case', () => {
+    let response = {};
+
+    beforeEach(async done => {
+      response = await supertest(app)
+        .post(postUsersPath)
+        .set(...defaultHeaders)
+        .send(usersTestData.createUserRequestBody);
+
+      done();
+    });
+
+    test('Should return a created status code', () => {
+      const { status } = response;
+
+      expect(status).toBe(201);
+    });
+
+    test('Should return the created user info without password', () => {
+      const { body } = response;
+      const userInfo = dbHelper.cleanRecord(body, true);
+
+      expect(userInfo).toMatchObject(usersTestData.expectedUserInfo);
+    });
+
+    test('Should save the user with the correct information', async done => {
+      const {
+        body: { id }
+      } = response;
+
+      const foundUser = await dbHelper.findUserById(id);
+      const cleanedFoundUser = dbHelper.cleanRecord(foundUser);
+
+      delete cleanedFoundUser.password;
+
+      expect(cleanedFoundUser).toStrictEqual({ id, ...usersTestData.expectedUserInfo });
+      done();
+    });
+  });
+
+  describe('Should fail correctly with schema error cases', () => {
+    const schemaErrorStatus = 422;
+
+    describe('Should throw an error if the password does not fullfil the schema', () => {
+      test('Should throw an error if the password length is lower than 8', async done => {
+        const shortPassword = '1234';
+
+        const { status, body } = await supertest(app)
+          .post(postUsersPath)
+          .set(...defaultHeaders)
+          .send({ ...usersTestData.createUserRequestBody, password: shortPassword });
+
+        expect(status).toBe(schemaErrorStatus);
```

Comment:
I suggest you to put each expect in an individually test in order to have more readability of the code :)